pkg/virt-handler/vm.go

```diff
@@ -2805,6 +2816,10 @@ func (d *VirtualMachineController) vmUpdateHelperDefault(origVMI *v1.VirtualMach
 		if err := d.getMemoryDump(vmi); err != nil {
 			return err
 		}
+
+		if err := d.hotplugVirtioInterfaces(vmi); err != nil {
+			log.Log.Object(vmi).Error(err.Error())
```

Comment:
> Please add a comment explaining why you are just logging and not returning an error. I guess you don't want a re-enquque. On startup we prevent re-attempt by returning a critical error. How is it prevented in the case of hotplug interface?

For now I'll just stop re en-queuing on a critical net error.

I'm aware this will again be re-tried once there's another update, but that'll only fail once, then stop.

I'll tackle identifying broken hot-plugged interfaces as a follow-up.

> 
> In general, it would be nice if the commit message would contain a small design explaining -
> 
> 1. How virt-handler knows the interface should be hotplugged?
> 2. How virt-handler differentiate between interfaces that were there since the startup, interfaces that were hotplugged and interfaces that have to be hotplugged.
> 3. What should be done if the hotplug failed on each stage (discover, prepare, libvirt, etc)?

Sure, makes sense. 

I'll update the commit messages for the entire PR later on - for now, I'm still focused in the code.

Reply:
I'm missing something: what will cause it to give up? Will this work indefinitely?
Comparing to the SR-IOV case, there we do have a backoff in place with a limit if I am not mistaken.

Reply:
Yes, it'll try indefinitely to plumb the new interface in.

This is useful for some scenarios - on the handler, I've only seen this scenario so far - you do not have enough pcie controller to plug the interface; it'll fail until you unplug something from the VMI, and then it'll succeed.

The above example is not possible in this PR since I removed unplug from here, but I'm just stating an example.

I'm sure there are more examples of recoverable errors I am not seeing right now.